In [1]:
import pandas as pd

In [2]:
sad_children = pd.read_csv('../cleaned_data/global_sad_children_2003_2015_ages_13_17_clean.csv')
internet = pd.read_csv('../cleaned_data/global_internet_use_1960_2022_clean.csv')

In [4]:
sad_children.head()

Unnamed: 0  Year LocationAbbr     LocationDesc          Topic  \
0      132659  2003          CNA  China (Beijing)  Mental Health   
1      189964  2003          CNA  China (Beijing)  Mental Health   
2      192210  2003          CNA  China (Beijing)  Mental Health   
3      250453  2003          CNA  China (Beijing)  Mental Health   
4      270912  2003          CNA  China (Beijing)  Mental Health   

        Subtopic    ShortQuestionText  \
0  Mental Health      Could not sleep   
1  Mental Health          Felt lonely   
2  Mental Health        Close Friends   
3  Mental Health  Made a suicide plan   
4  Mental Health   Considered suicide   

                               Greater_Risk_Question  \
0  Most of the time or always were so worried abo...   
1             Most of the time or always felt lonely   
2                         Did not have close friends   
3   Made a plan about how they would attempt suicide   
4            Seriously considered attempting suicide   

                              Description    Age    Sex Data_Value_Symbol  \
0  during the 12 months before the survey  Total  Total                 %   
1  during the 12 months before the survey  Total  Total                 %   
2                                     NaN  Total  Total                 %   
3  during the 12 months before the survey  Total  Total                 %   
4  during the 12 months before the survey  Total  Total                 %   

  Data_Value_Type  Greater_Risk_Data_Value  
0      Percentage                   4.5379  
1      Percentage                   6.6563  
2      Percentage                   7.6368  
3      Percentage                   6.7155  
4      Percentage                  14.2575

In [7]:
sad_children = sad_children.drop(sad_children.columns[0], axis = 1)
sad_children.columns

Index(['Year', 'LocationAbbr', 'LocationDesc', 'Topic', 'Subtopic',
       'ShortQuestionText', 'Greater_Risk_Question', 'Description', 'Age',
       'Sex', 'Data_Value_Symbol', 'Data_Value_Type',
       'Greater_Risk_Data_Value'],
      dtype='object')

In [9]:
sad_children['LocationDesc'] = sad_children['LocationDesc'].str.split('(').str[0]
sad_children['LocationDesc'] = sad_children['LocationDesc'].str.split('-').str[0]

In [5]:
internet.head()

Country Name  1960  1961  1962  1963  1964  1965  1966  \
0                        Aruba   0.0   0.0   0.0   0.0   0.0   0.0   0.0   
1  Africa Eastern and Southern   0.0   0.0   0.0   0.0   0.0   0.0   0.0   
2                  Afghanistan   0.0   0.0   0.0   0.0   0.0   0.0   0.0   
3   Africa Western and Central   0.0   0.0   0.0   0.0   0.0   0.0   0.0   
4                       Angola   0.0   0.0   0.0   0.0   0.0   0.0   0.0   

   1967  1968  ...       2012       2013       2014       2015       2016  \
0   0.0   0.0  ...  74.000000  78.900000  83.780000  88.661227  93.542454   
1   0.0   0.0  ...   8.534750  10.126945  12.237716  14.593148  16.378025   
2   0.0   0.0  ...   5.454545   5.900000   7.000000   8.260000  11.000000   
3   0.0   0.0  ...  10.293655  12.577756  14.923441  18.179203  21.829266   
4   0.0   0.0  ...   7.700000  13.000000  21.400000  22.000000  23.200000   

        2017       2018       2019       2020       2021  
0  97.170000  97.170000  97.170000  97.170000  97.170000  
1  18.098060  21.653313  24.247500  26.200657  26.200657  
2  13.500000  16.800000  17.600000  18.400000  18.400000  
3  25.176342  29.305247  29.023474  34.093121  34.093121  
4  26.000000  29.000000  32.400000  36.000000  36.000000  

[5 rows x 63 columns]

In [8]:
internet = internet.melt(
    id_vars=['Country Name'], 
    var_name='Year', 
    value_name='Percent Internet Use'
)

internet.head()

Country Name  Year  Percent Internet Use
0                        Aruba  1960                   0.0
1  Africa Eastern and Southern  1960                   0.0
2                  Afghanistan  1960                   0.0
3   Africa Western and Central  1960                   0.0
4                       Angola  1960                   0.0

In [11]:
internet.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 16492 entries, 0 to 16491
Data columns (total 3 columns):
 #   Column                Non-Null Count  Dtype  
---  ------                --------------  -----  
 0   Country Name          16492 non-null  object 
 1   Year                  16492 non-null  object 
 2   Percent Internet Use  16492 non-null  float64
dtypes: float64(1), object(2)
memory usage: 386.7+ KB


In [13]:
sad_children.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 1313 entries, 0 to 1312
Data columns (total 13 columns):
 #   Column                   Non-Null Count  Dtype  
---  ------                   --------------  -----  
 0   Year                     1313 non-null   int64  
 1   LocationAbbr             1313 non-null   object 
 2   LocationDesc             1313 non-null   object 
 3   Topic                    1313 non-null   object 
 4   Subtopic                 1313 non-null   object 
 5   ShortQuestionText        1313 non-null   object 
 6   Greater_Risk_Question    1313 non-null   object 
 7   Description              1073 non-null   object 
 8   Age                      1313 non-null   object 
 9   Sex                      1313 non-null   object 
 10  Data_Value_Symbol        1313 non-null   object 
 11  Data_Value_Type          1313 non-null   object 
 12  Greater_Risk_Data_Value  1313 non-null   float64
dtypes: float64(1), int64(1), object(11)
memory usage: 133.5+ KB


In [15]:
sad_children['Year'] = sad_children['Year'].astype(str)

In [16]:
globe_internet_childsad = pd.merge(sad_children, internet, 
                               left_on = ('Year', 'LocationDesc'), right_on = ('Year', 'Country Name'), 
                               how = 'inner')

In [17]:
globe_internet_childsad

Year LocationAbbr LocationDesc          Topic       Subtopic  \
0    2003           KE        Kenya  Mental Health  Mental Health   
1    2003           KE        Kenya  Mental Health  Mental Health   
2    2003           KE        Kenya  Mental Health  Mental Health   
3    2003           KE        Kenya  Mental Health  Mental Health   
4    2003           KE        Kenya  Mental Health  Mental Health   
..    ...          ...          ...            ...            ...   
440  2015           TH     Thailand  Mental Health  Mental Health   
441  2015           TH     Thailand  Mental Health  Mental Health   
442  2015           TH     Thailand  Mental Health  Mental Health   
443  2015           TH     Thailand  Mental Health  Mental Health   
444  2015           TH     Thailand  Mental Health  Mental Health   

       ShortQuestionText                              Greater_Risk_Question  \
0            Felt lonely             Most of the time or always felt lonely   
1        Could not sleep  Most of the time or always were so worried abo...   
2    Made a suicide plan   Made a plan about how they would attempt suicide   
3          Close Friends                         Did not have close friends   
4     Considered suicide            Seriously considered attempting suicide   
..                   ...                                                ...   
440    Attempted suicide                                  Attempted suicide   
441          Felt lonely             Most of the time or always felt lonely   
442  Made a suicide plan   Made a plan about how they would attempt suicide   
443      Could not sleep  Most of the time or always were so worried abo...   
444   Considered suicide            Seriously considered attempting suicide   

                                           Description    Age    Sex  \
0               during the 12 months before the survey  Total  Total   
1               during the 12 months before the survey  Total  Total   
2               during the 12 months before the survey  Total  Total   
3                                                  NaN  Total  Total   
4               during the 12 months before the survey  Total  Total   
..                                                 ...    ...    ...   
440  one or more times during the 12 months before ...  Total  Total   
441             during the 12 months before the survey  Total  Total   
442             during the 12 months before the survey  Total  Total   
443             during the 12 months before the survey  Total  Total   
444             during the 12 months before the survey  Total  Total   

    Data_Value_Symbol Data_Value_Type  Greater_Risk_Data_Value Country Name  \
0                   %      Percentage                  17.4709        Kenya   
1                   %      Percentage                  15.5423        Kenya   
2                   %      Percentage                  30.0063        Kenya   
3                   %      Percentage                  11.6787        Kenya   
4                   %      Percentage                  27.9374        Kenya   
..                ...             ...                      ...          ...   
440                 %      Percentage                  12.9528     Thailand   
441                 %      Percentage                   9.7469     Thailand   
442                 %      Percentage                  13.7552     Thailand   
443                 %      Percentage                   8.8662     Thailand   
444                 %      Percentage                  12.2252     Thailand   

     Percent Internet Use  
0                2.941903  
1                2.941903  
2                2.941903  
3                2.941903  
4                2.941903  
..                    ...  
440             39.316127  
441             39.316127  
442             39.316127  
443             39.316127  
444             39.316127  

[445 rows x 15 columns]

In [18]:
globe_internet_childsad.to_csv('Global_internet_sadchildren.csv', index = False)

In [19]:
globe_internet_childsad.corr()

Greater_Risk_Data_Value  Percent Internet Use
Greater_Risk_Data_Value                 1.000000             -0.073014
Percent Internet Use                   -0.073014              1.000000

#### correlation coefficient between internet users and sad children is -0.073